Include/pyport.h

```diff
@@ -510,6 +510,7 @@ extern "C" {
  *    Py_DEPRECATED(3.4) typedef int T1;
  *    Py_DEPRECATED(3.8) PyAPI_FUNC(int) Py_OldFunction(void);
  */
+#ifndef Py_DEPRECATED
```

Comment:
Why do you modify pyport.h?

Reply:
IMHO it's fine if compiling _testcapimodule.c emits a deprecation warning. If it becomes an issue, we can just remove the test.

Reply:
> IMHO it's fine if compiling _testcapimodule.c emits a deprecation warning.

The more warnings are shown, the less value there is in those warnings. A single warning during compilation stands out, an additional warning in between many not. So we should strive for compiling CPython without any warnings.

> If it becomes an issue, we can just remove the test.

I think that's the worst solution. We typically test deprecated functionality in pure Python modules, why shouldn't we test deprecated C API?

Reply:
Reading https://stackoverflow.com/questions/13459602/how-can-i-get-rid-of-deprecated-warnings-in-deprecated-functions-in-gcc/ suggests there are a few ways we can go here:

1. Mark the _testcapi init function itself as deprecated, so compilers won't care that it accesses deprecated symbols.
2. Start defining compiler specific `_Py_PUSH_PRAGMA_IGNORE_DEPRECATION_WARNINGS` and `_Py_POP_PRAGMA` macros and use those to selectively silence compiler warnings
3. Use the testcapi warnings to check that deprecated C APIs actually *are* deprecated (doing that long term might involve pulling out a separately compiled module for those though, so we can more easily distinguish them from actual warnings in the main interpreter and standard library build process)

For right now, I'd suggest seeing if the warning goes away if you mark the testcapi module's init function as deprecated.